tests/integration_tests/test_validate_output.py

```diff
@@ -4,34 +4,33 @@
 import os.path
 import subprocess
 from bs4 import BeautifulSoup
-import itertools
 
 
 # Run pipeline in test mode, this is done once per test session
 @pytest.fixture(scope="session", autouse=True)
-def run_pipeline(tmpdir_factory):
+def result_dir(request, tmpdir_factory):
+    demultiplexer = "bcl2fastq"
+    if hasattr(request, "param"):
+        demultiplexer = request.param
+
     result_dir = tmpdir_factory.mktemp("results")
-    subprocess.run(
-        [
-            "nextflow",
-            "run",
-            "main.nf",
-            "-profile",
-            "dev,test,singularity",
-            "--result_dir",
-            result_dir,
-        ],
-        check=True,
-    )
-    yield result_dir
+    cmd = [
+        "nextflow",
+        "run",
+        "main.nf",
+        "-profile",
+        "dev,test,singularity",
+        "--demultiplexer",
+        demultiplexer,
+        "--result_dir",
+        result_dir,
+    ]
+    if demultiplexer == "bclconvert":
+        cmd = [*cmd, *["--run_folder", "test_data/230825_M04034_0043_000000000-L6NVV"]]
```

Comment:
I think it would be cleaner to create another test profile instead: https://github.com/Molmed/seqreports/blob/f49a23e652af5859b43c8426a28ba32693f00148/nextflow.config#L62
It could for exmpale be named `test_bclconvert` which points to a config with that runfolder set.